kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/extension/VssSpecificationCopyExtension.kt

```diff
@@ -137,17 +148,38 @@ fun <T : VssSpecification> T.copy(
         .find { it.vssPath == vssPath } ?: return this
 
     val updatedVssProperty = vssProperty.copy(updatedValue)
-    val relevantChildren = findHeritageLine(vssProperty).toMutableList()
 
-    // Replace the last specification (Property) with the changed one
-    val updatedVssSpecification: T = if (relevantChildren.isNotEmpty()) {
-        relevantChildren.removeLast()
-        relevantChildren.add(updatedVssProperty)
+    // Same property with no heirs, no deep copy is needed
+    if (this.vssPath == updatedVssProperty.vssPath) return updatedVssProperty as T
 
-        this.deepCopy(relevantChildren)
-    } else {
-        updatedVssProperty as T // The property must be T since no children are available
-    }
+    return deepCopy(0, updatedVssProperty)
+}
 
-    return updatedVssSpecification
+/**
+ * Convenience operator for [copy] with a value [T].
+ */
+operator fun <T : Any> VssProperty<T>.plusAssign(value: T) {
+    copy(value)
+}
+
+/**
+ * Convenience operator for [copy] with a value [T].
+ */
+operator fun <T : Any> VssProperty<T>.plus(value: T): VssProperty<T> {
+    return copy(value)
+}
+
+/**
+ * Convenience operator for [copy] with a [Boolean] value which will be inverted.
+ */
+operator fun VssProperty<Boolean>.not(): VssProperty<Boolean> {
+    return copy(!value)
```

Comment:
I will handle the copy method gracefully now and don't update the properties if an error occurs.

Reply:
I'm unsure if this is really a good idea, since our models can get pretty complex it can be really hard to verify (by sight) if my changes to be expected were correctly propagated and therefore I would expect it to fail-fast if any issue appeared instead of gracefully ignoring it - but of course that's a philosophical topic

Reply:
Reverted back to throwing an exception